content/docs/hooks-overview.md

```diff
@@ -1,107 +1,107 @@
+Cú pháp [array destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring) cho phép chúng ta dùng các tên khác nhau cho các biến trạng thái khi khai báo gọi hàm `useState`. Các tên đó không thuộc về `useState` API. Thay vì thế, React giả sử nếu bạn gọi `useState` nhiều lần, bạn làm thế trên cùng thứ tự mọi lần render. Chúng tôi sẽ giải thích tại sao cái này hoạt động và khi nào nó hữu ích sau.
 
-#### But what is a Hook? {#but-what-is-a-hook}
+#### Nhưng Hook là gì? {#but-what-is-a-hook}
 
-Hooks are functions that let you “hook into” React state and lifecycle features from function components. Hooks don't work inside classes -- they let you use React without classes. (We [don't recommend](/docs/hooks-intro.html#gradual-adoption-strategy) rewriting your existing components overnight but you can start using Hooks in the new ones if you'd like.)
+Hooks là các hàm mà cho phép bạn "hook into (móc vào)" trạng thái của React và các tính năng vòng đời  từ các hàm components. Hooks không hoạt động bên trong classes -- chúng cho phép bạn sử dụng React không cần classes. (Chúng tôi [không khuyến khích](/docs/hooks-intro.html#gradual-adoption-strategy) viết lại các components hiện tại của bạn qua đêm nhưng bạn có thể sử dụng Hooks trong những cái mới nếu bạn thích.)
 
-React provides a few built-in Hooks like `useState`. You can also create your own Hooks to reuse stateful behavior between different components. We'll look at the built-in Hooks first.
+React cung cấp một vài Hooks sẵn có như `useState`. Bạn cũng có thể tạo Hooks của bạn để sử dụng lại những hành vi có trạng thái giữa các components khác nhau. Chúng tôi sẽ xem các Hooks có sẵn trước.
 
->Detailed Explanation
+>Diễn giải chi tiết
 >
->You can learn more about the State Hook on a dedicated page: [Using the State Hook](/docs/hooks-state.html).
+>Bạn có thể học thêm về State Hook tại: [Sử dụng State Hook](/docs/hooks-state.html).
 
 ## ⚡️ Effect Hook {#effect-hook}
 
-You've likely performed data fetching, subscriptions, or manually changing the DOM from React components before. We call these operations "side effects" (or "effects" for short) because they can affect other components and can't be done during rendering.
+Bạn thực hiện lấy dữ liệu, đăng ký, hoặc thay đổi DOM thủ công từ React components trước đây. Chúng tôi gọi các hoạt động đó là "side effects" (hoặc "effects" cho ngắn) bởi vì chúng có thể ảnh hưởng các components khác và không để xong trong qua trình render.
```

Comment:
can't be done during rendering should be translated to "không thể thực hiện trong quá trình render"

note the typo "qua" here also